src/diffpy/srmise/applications/extract.py

```diff
@@ -419,82 +419,87 @@ def main():
     from diffpy.srmise.pdfpeakextraction import PDFPeakExtraction
     from diffpy.srmise.srmiseerrors import SrMiseDataFormatError, SrMiseFileError
 
-    try:
-        options.peakfunction = eval("peaks." + options.peakfunction)
-    except Exception as err:
-        print(err)
-        print("Could not create peak function '%s'. Exiting." % options.peakfunction)
-        return
-
-    try:
-        options.modelevaluator = eval("modelevaluators." + options.modelevaluator)
-    except Exception as err:
-        print(err)
-        print("Could not find ModelEvaluator '%s'. Exiting." % options.modelevaluator)
-        return
-
-    if options.bcrystal is not None:
+    if options.peakfunction:
```

Comment:
Ensure the `try` block is only executed when the `options.peakfunction` is set to `true` (bug fix)